### PR TITLE
add h5py to dockerfile

### LINKED ISSUE #11356: Cannot run example in tensorflow docker due to missing dependency
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 14.04
- **TensorFlow installed from (source or binary)**: running in official docker container
- **TensorFlow version (use command below)**: 1.1.0
- **Python version**: 2.7
- **CUDA/cuDNN version**: CUDA 8.0
- **GPU model and memory**: Titan X
- **Exact command to reproduce**:
`docker run -it -p 8888:8888 gcr.io/tensorflow/tensorflow`

### Describe the problem
The official tensorflow docker container does not include h5py. This might be intentional?

However, this breaks an example:
- tensorflow/tensorflow/examples/learn/hdf5_classification.py

This also causes a number of contrib methods to raise exceptions:
- tensorflow/tensorflow/contrib/learn/python/learn/learn_io/data_feeder_test.py
- a numbert of keras.contrib files

### PR DESCRIPTION
**Pull request by yhenon**
Add h5py to dockerfile. Fixes #11356

### CODE REVIEW

**Comment by yhenon** [comment]
I signed it!

File: tensorflow/tools/docker/Dockerfile [hunk 1]
```diff
@@ -32,6 +32,7 @@ RUN pip --no-cache-dir install \
         sklearn \
         pandas \
         Pillow \
+        h5py \
```
Comment (caisq) [inline tensorflow/tools/docker/Dockerfile]: Keep the list alphabetically sorted? Same below.

**Review by caisq — CHANGES_REQUESTED** [review]

**Comment by yhenon** [comment]
Fixed as requested. The list wasn't alphabetical to start with, so I fixed that too.

**Comment by caisq** [comment]
Thanks, @yhenon

**Review by caisq — APPROVED** [review]

**Comment by caisq** [comment]
PR merged. The changes will take effect in the tensorflow/tensorflow release images after the next release and in the tensorflow/tensorflow:nightly* images after the next nightly build.

**Pull request merged**